src/python/zquantum/core/hamiltonian.py

```diff
@@ -200,49 +216,18 @@ def get_expectation_values_from_rdms(
     reordered_qubitoperator = QubitOperator()
     for term, coefficient in terms_iterator:
         reordered_qubitoperator += QubitOperator(term, coefficient)
-
     expectations_packed = interactionrdm.get_qubit_expectations(reordered_qubitoperator)
 
-    if () in expectations_packed.terms:
-        del expectations_packed.terms[
-            ()
-        ]  # Expectation of the constant term is excluded from expectation values
-
     expectations = np.array(list(expectations_packed.terms.values()))
     if np.any(np.abs(np.imag(expectations)) > 1e-3):
         raise RuntimeWarning(
             f"Expectation values extracted from rdms inside get_expectation_values_from_rdms are complex!"
         )
     expectations = np.real(expectations)
     np.clip(expectations, -1, 1, out=expectations)
-
     return ExpectationValues(expectations)
 
 
-def estimate_nmeas_for_operator(
```

Comment:
It won't solve the problem, but it gets rid of a pathway that's very prone to user error.

Since the function has `ExpectationValues` and `QubitOperator` as parameters, one might naturally expect that the `ExpectationValues` should be in the same order as the corresponding terms of the `QubitOperator`. But this will actually give the wrong result because the expectation values need to be ordered according to how the terms are ordered after grouping.

@mstechly and I discovered that in fact some of the unit tests were making this mistake, but they just happened to edge cases where grouping didn't change the order!